core/sequencer/sequencer_api.proto

```diff
@@ -71,19 +58,66 @@ message RunBatchRequest {
   int32 max_batch = 3;
 }
 
-// PublishBatchRequest copies all SignedMapHeads into the Log of SignedMapHeads.
-message PublishBatchRequest {
+// DefineRevisionRequest contains information needed to define a new revision.
+message DefineRevisionsRequest {
+  // directory_id is the directory to examine the outstanding mutations for.
   string directory_id = 1;
+  // min_batch is the minimum number of items in a batch.
+  // If less than min_batch items are available, nothing happens.
+  // TODO(#1047): Replace with timeout so items in the log get processed
+  // eventually.
+  int32 min_batch = 2;
+  // max_batch is the maximum number of items in a batch.
+  int32 max_batch = 3;
+}
+
+// DefineRevisionResponse contains information about freshly defined revisions.
+message DefineRevisionsResponse {
+  // outsanding_revisions a list of all the defined revisions which are not yet applied.
+  repeated int64 outstanding_revisions = 1;
+}
+
+// ApplyRevisionRequest contains information needed to create a new revision.
+message ApplyRevisionRequest {
+  // directory_id is the directory to apply the mutations to.
+  string directory_id = 1;
+  // revision is the expected revision of the new revision.
+  int64 revision = 2;
+}
+
+// ApplyRevisionResponse contains stats about the created revision.
+message ApplyRevisionResponse {
+  string directory_id = 1;
+  // The revision this is for.
+  int64 revision = 2;
+  // mutations processed.
+  int64 mutations = 3;
+  // map_leaves written.
+  int64 map_leaves = 4;
+}
+
+// PublishRevisionRequest copies all SignedMapHeads into the Log of SignedMapHeads.
+message PublishRevisionsRequest {
+  string directory_id = 1;
+}
+
+// PublishRevisions
+message PublishRevisionsResponse {
+  // revisions published.
+  repeated int64 revisions = 1;
 }
 
 // The KeyTransparency Sequencer API.
 service KeyTransparencySequencer {
   // RunBatch reads outstanding mutations and calls CreateRevision.
   rpc RunBatch(RunBatchRequest) returns (google.protobuf.Empty);
-  // CreateRevision applies the contained mutations to the current map root.
+  // DefineRevision examines the outstanding items in the queue and optionally
+  // commits to one or more revisions.
```

Comment:
> and optionally commits to one or more revisions

I'm not sure what committing to a revision means, can you clarify?

Reply:
Update requests go through the following process:

1. `QueueEntryUpdate` writes `EntryUpdate` to a *log of mutations* 
2. `DefineBatch` reads the highest timestamp in the log of mutations and decides that all the log items between `LowestWatermark` and `HighestWatermark` are going to be part of map revision `N`.  We write this to disk first so that we can retry it in case anything goes wrong. 
3. `ApplyRevision` reads all the log messages between low and high watermarks, decides which messages are valid, and writes all the information to the Trillian map at revision `N`. 
4. `PublishRevisions` takes all the signed map roots, including revision `N` and writes them to the Trillian log of signed map roots.  At this point, clients will be able to see map revision `N` and use the values inside of it. 

Each of these steps can be retried on its own if anything goes wrong.